src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -2,294 +2,450 @@
-           As such we can decide whether tx is well formed but we also have to validate nonce
-           just before the execution of the block / tx. */
-        public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec)
+    public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec) => IsWellFormed(transaction, releaseSpec, out _);
+
+    /// <summary>
+    /// Full and correct validation is only possible in the context of a specific block
+    /// as we cannot generalize correctness of the transaction without knowing the EIPs implemented
+    /// and the world state(account nonce in particular).
+    /// Even without protocol change, the tx can become invalid if another tx
+    /// from the same account with the same nonce got included on the chain.
+    /// As such, we can decide whether tx is well formed as long as we also validate nonce
+    /// just before the execution of the block / tx.
+    /// </summary>
+    public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec, out string? error)
+    {
+        if (!_validators.TryGetValue(transaction.Type, out ITxValidator? validator))
         {
-            return IsWellFormed(transaction, releaseSpec, out _);
-        }
-        public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec, out string? error)
-        {
-            error = null;
-
-            // validate type before calculating intrinsic gas to avoid exception
-            return ValidateTxType(transaction, releaseSpec, ref error)
-                   // This is unnecessarily calculated twice - at validation and execution times.
-                   && ValidateWithError(transaction.GasLimit >= IntrinsicGasCalculator.Calculate(transaction, releaseSpec), TxErrorMessages.IntrinsicGasTooLow, ref error)
-                   // if it is a call or a transfer then we require the 'To' field to have a value while for an init it will be empty
-                   && ValidateWithError(ValidateSignature(transaction, releaseSpec), TxErrorMessages.InvalidTxSignature, ref error)
-                   && ValidateChainId(transaction, ref error)
-                   && ValidateWithError(Validate1559GasFields(transaction, releaseSpec), TxErrorMessages.InvalidMaxPriorityFeePerGas, ref error)
-                   && ValidateWithError(Validate3860Rules(transaction, releaseSpec), TxErrorMessages.ContractSizeTooBig, ref error)
-                   && Validate4844Fields(transaction, ref error);
+            error = TxErrorMessages.InvalidTxType(releaseSpec.Name);
+            return false;
         }
 
-        private static bool Validate3860Rules(Transaction transaction, IReleaseSpec releaseSpec) =>
-            !transaction.IsAboveInitCode(releaseSpec);
+        return validator.IsWellFormed(transaction, releaseSpec, out error);
+    }
+}
+
+public sealed class AllTxValidator(List<ITxValidator> validators) : ITxValidator
```

Comment:
`CompositeTxValidator` (we use the term `Composite` in multiple places already for this pattern)

Reply:
I went with `AllTxValidator` since you could have a validator that checks that at least one validator returns `true`, that is, merge validations using `||` instead of `&&`, but I guess we don't have a use case for that.
Renamed to `CompositeTxValidator` for consistency.